### PR TITLE
LPS-146764 Export depot connecitons between staging and live

### PR DESCRIPTION
**Pull request by adolfopa**
The column `groupId` in the group rels pointed to the site group ID, not the depot group ID. When staging/ct looks for changes to export/import, it will look for models that have a group ID equal to the group ID of the exporting/importing group. Originally, the group ID column contained the group of the site, so staging/ct interpreted that as if the group rel belonged to the site, and not the depot (in other words, the rel would be exported/imported when you tried to export/import the site).

This PR adds change tracking support for completeness (looks like an oversight not to support that), and fixes the values of the group ID column so that it points to the depot group.

### CODE REVIEW

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by Hanlf** [comment]
Hi @adolfopa 
There are 1 test fix and 3 portal issues after analyzing the depot tests result, let me know if you need more info or any other help.

- For the Poshi case issue, could you please fetch the last commit(8247a44) of my [PR](https://github.com/Hanlf/liferay-portal/pull/310) first when you fix the portal issue? I prefer to use your original PR rather than opening a new PR by me, thanks!

- For the portal issues, please see the below for details:

**Issue 1: Asset library cannot be selected on connected site when staging is enabled**
Steps:
1. Add a new site
2. Add a depot and connect to the added site
3. Enable local staging for the site and depot
4. Upload an image to Documents and Media in the depot 
5. Go to the staging site > Blogs
6. Add a blog entry and try to select the image from the depot
**Result:** The depot not showing in the item selector

**Issue 2: Show the depot name in the Connected sites table after enabling local staging**
Steps:
1. Add a new site
2. Add a depot and connect to the added site
3. Go to Asset Library Settings > Sites to assert the Name is site name in the Connected sites
4. Enable local staging for the depot
5. Go to Asset Library Settings > Sites to assert the Name column
**Result:** The name is updated to depot name, instead of the site name

**Issue 3: Fail to publish documents to live on the site when using DM type from depot**
Steps:
1. Enable remote staging for depot and site
2. Go to Asset Library Settings > Sites to Make Structures Available
3. Add a new DM type in the depot
4. Publish it to live in the depot
5. Go to the staging site > DM to add a new document using the new DM type from the depot
6. Try to publish it to live
**Result:** Fail to publish documents to live on the site for remote staging
**Note:** not sure if local staging has the same issue, which is blocked by another issue: no depot document type shows on the connected site, I guess should be the same root cause of Issue 2.
Error throws on the console:
```
2022-02-07 06:59:39.947 ERROR [liferay/background_task-9][LoggerExportImportLifecycleListener:196] Layout remote publication failed for group 43649
2022-02-07 06:59:40.072 ERROR [liferay/background_task-9][BackgroundTaskMessageListener:182] Unable to execute background task
com.liferay.portal.kernel.exception.PortalException: Invocation failed due to com.liferay.exportimport.kernel.lar.PortletDataException
	at com.liferay.portal.servlet.TunnelServlet.doPost(TunnelServlet.java:117) ~[portal-impl.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:681) ~[servlet-api.jar:4.0.FR]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:764) ~[servlet-api.jar:4.0.FR]
	at com.liferay.shielded.container.internal.proxy.ServletWrapper.service(ServletWrapper.java:113) ~[com.liferay.shielded.container.impl.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:227) ~[catalina.jar:9.0.56]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.56]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:53) ~[tomcat-websocket.jar:9.0.56]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.56]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.56]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:124) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.processFilter(BaseFilter.java:147) ~[portal-kernel.jar:?]
	at com.liferay.portal.servlet.filters.authverifier.AuthVerifierFilter.processFilter(AuthVerifierFilter.java:212) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:49) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:215) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116) ~[portal-kernel.jar:?]
	at com.liferay.portal.servlet.filters.password.modified.PasswordModifiedFilter.processFilter(PasswordModifiedFilter.java:62) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:49) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:215) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116) ~[portal-kernel.jar:?]
	at com.liferay.portal.servlet.filters.lockout.LockoutFilter.processFilter(LockoutFilter.java:58) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:49) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:215) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.processFilter(BaseFilter.java:147) ~[portal-kernel.jar:?]
	at com.liferay.portal.sharepoint.SharepointFilter.processFilter(SharepointFilter.java:88) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:49) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:215) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.processFilter(BaseFilter.java:147) ~[portal-kernel.jar:?]
	at com.liferay.portal.servlet.filters.virtualhost.VirtualHostFilter.processFilter(VirtualHostFilter.java:245) ~[portal-impl.jar:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:49) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:215) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:196) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:99) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:196) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:99) ~[portal-kernel.jar:?]
	at org.tuckey.web.filters.urlrewrite.RuleChain.handleRewrite(RuleChain.java:176) ~[urlrewritefilter.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.RuleChain.doRules(RuleChain.java:145) ~[urlrewritefilter.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriter.processRequest(UrlRewriter.java:92) ~[urlrewritefilter.jar:4.0.4]
	at org.tuckey.web.filters.urlrewrite.UrlRewriteFilter.doFilter(UrlRewriteFilter.java:389) ~[urlrewritefilter.jar:4.0.4]
	at com.liferay.portal.url.rewrite.filter.internal.URLRewriteFilter.processFilter(URLRewriteFilter.java:98) ~[?:?]
	at com.liferay.portal.kernel.servlet.BaseFilter.doFilter(BaseFilter.java:49) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDoFilter(InvokerFilterChain.java:215) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:116) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:175) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:99) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:175) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:99) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.processDirectCallFilter(InvokerFilterChain.java:196) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilterChain.doFilter(InvokerFilterChain.java:99) ~[portal-kernel.jar:?]
	at com.liferay.portal.kernel.servlet.filters.invoker.InvokerFilter.doFilter(InvokerFilter.java:104) ~[portal-kernel.jar:?]
	at com.liferay.shielded.container.internal.proxy.FilterWrapper.doFilter(FilterWrapper.java:79) ~[com.liferay.shielded.container.impl.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:189) ~[catalina.jar:9.0.56]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:162) ~[catalina.jar:9.0.56]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:197) ~[catalina.jar:9.0.56]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:97) ~[catalina.jar:9.0.56]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:540) ~[catalina.jar:9.0.56]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:135) ~[catalina.jar:9.0.56]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:92) ~[catalina.jar:9.0.56]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:78) ~[catalina.jar:9.0.56]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:357) ~[catalina.jar:9.0.56]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:382) ~[tomcat-coyote.jar:9.0.56]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) ~[tomcat-coyote.jar:9.0.56]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:895) ~[tomcat-coyote.jar:9.0.56]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1732) ~[tomcat-coyote.jar:9.0.56]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) ~[tomcat-coyote.jar:9.0.56]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191) ~[org.apache.tomcat-util.jar:9.0.56]
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659) ~[org.apache.tomcat-util.jar:9.0.56]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) ~[org.apache.tomcat-util.jar:9.0.56]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_181]
```

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

File: modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryGroupRelLocalServiceImpl.java [hunk 1]
```diff
@@ -53,7 +56,7 @@ public DepotEntryGroupRel addDepotEntryGroupRel(
 		depotEntryGroupRel = depotEntryGroupRelPersistence.create(
 			counterLocalService.increment());
 
-		depotEntryGroupRel.setGroupId(toGroupId);
+		depotEntryGroupRel.setGroupId(_getDepotGroupId(depotEntryId));
```
[the remark this answers is not in the document]
Comment (adolfopa) [reply modules/apps/depot/depot-service/src/main/java/com/liferay/depot/service/impl/DepotEntryGroupRelLocalServiceImpl.java:59]: Thanks Alicia. I'm going to skip the upgrade; we'll have to keep the groupId = depotGroupId association. There are more places that assume that, and I don't feel comfortable with changing so much stuff. So, we'll let sites "own" the connection and document that as a known behaviour.

**Comment by adolfopa** [comment]
I've pushed the changes into https://github.com/liferay-lima/liferay-portal/pull/2712; let's see if everything is green there. Thanks @Hanlf and @AliciaGarciaGarcia !

**Pull request closed without merging**